common/compositor/va/varenderer.cpp

```diff
@@ -440,18 +448,33 @@ bool VARenderer::UpdateCaps() {
       if (use_default) {
         itr->second.value_ = itr->second.caps_.range.default_value;
       }
-      cbparam.type = VAProcFilterColorBalance;
-      cbparam.value = itr->second.value_;
-      cbparam.attrib = itr->second.caps_.type;
-      if (!cb_elements[static_cast<int>(itr->first)].CreateBuffer(
-              va_context_, VAProcFilterParameterBufferType,
-              sizeof(VAProcFilterParameterBufferColorBalance), 1, &cbparam)) {
-        return false;
-      }
-      filters_.push_back(cb_elements[static_cast<int>(itr->first)].buffer());
+      cbparam[index].type = VAProcFilterColorBalance;
+      cbparam[index].value = itr->second.value_;
+      cbparam[index].attrib = itr->second.caps_.type;
+      index++;
     }
   }
 
+  // workarround if filter num is 0, then set HSBC/Deinterlace will meet green
+  // screen.
+  if (!index) {
+    cbparam[index].type = VAProcFilterColorBalance;
+    cbparam[index].value = colorbalance_caps_[HWCColorControl::kColorHue]
+                               .caps_.range.default_value;
+    cbparam[index].attrib =
+        colorbalance_caps_[HWCColorControl::kColorHue].caps_.type;
+    index++;
+  }
+
+  if (index) {
```

Comment:
@feiwan1 since the **if** condition on line 460 will always cause index to be incremented on line 466 (if index is ever 0), would the other **if** condition on line 469 be necessary? Currently, it is possible for both **if** conditions to run `if index is 0`. Otherwise, if the intended effect is to have only one or the other we could change the **if** on line 469 to **else if** instead of removing it.